devlib/utils/types.py

```diff
@@ -68,6 +68,15 @@ def numeric(value):
     """
     if isinstance(value, int):
         return value
+
+    if isinstance(value, basestring):
+        value = value.strip()
+        if value.endswith('%'):
+            try:
+                return float(value.rstrip('%')) / 100
```

Comment:
If we ran into a percentage we don't want to drop through to the rest of the function after striping away the `%`. Say we run into `"10%"` afterwards, you end up with `"10"`. 

- If the logic is implemented exactly as you've suggested, then we will be returning `10` as in that case it is the `ivalue`, not the `fvalue` that  is returned.
- If we appy `corrector` to the 'ivalue' we always end up with a float.

The `corrector` can be changed to default to `1` rather than `1.0`, in which case we should get the correct behavior. However, that's rather subtle, and I would argue, less clear than the more explicit current approach. Additionally, if see a percentage, we will always want to return a float, therefore there is no point doing the subsequent integer conversion and comparison -- better to return right away.

Regarding factoring out the divisor into a variable -- I suggest we hold off until we're actually adding the support for per-mille or other such value...